cli/azd/pkg/tools/azcli/container_service.go

```diff
@@ -0,0 +1,51 @@
+package azcli
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerservice/armcontainerservice/v2"
+)
+
+type ContainerServiceClient interface {
+	GetAdminCredentials(
+		ctx context.Context,
+		resourceGroupName string,
+		resourceName string,
+	) (*armcontainerservice.CredentialResults, error)
+}
+
+type containerServiceClient struct {
+	client         *armcontainerservice.ManagedClustersClient
+	subscriptionId string
+}
+
+func NewContainerServiceClient(
+	subscriptionId string,
+	credential azcore.TokenCredential,
+	options *arm.ClientOptions,
+) (ContainerServiceClient, error) {
+	azureClient, err := armcontainerservice.NewManagedClustersClient(subscriptionId, credential, options)
+	if err != nil {
+		return nil, err
+	}
+
+	return &containerServiceClient{
+		subscriptionId: subscriptionId,
+		client:         azureClient,
+	}, nil
+}
+
+func (cs *containerServiceClient) GetAdminCredentials(
+	ctx context.Context,
+	resourceGroupName string,
+	resourceName string,
+) (*armcontainerservice.CredentialResults, error) {
+	creds, err := cs.client.ListClusterAdminCredentials(ctx, resourceGroupName, resourceName, nil)
```

Comment:
I am curious about why we are using `nil` for options here.

![image](https://user-images.githubusercontent.com/24213737/222304072-89b288ba-8f90-42c0-9be6-ffedb81463d8.png)

Can you add a note about why `ServerFqdn` is not relevant for us?

Reply:
I'm not familiar with the option and the azure docs are not much help.
https://learn.microsoft.com/en-us/rest/api/aks/managed-clusters/list-cluster-admin-credentials?tabs=HTTP